src/library_wasm_worker.js

```diff
@@ -182,7 +182,7 @@ if (ENVIRONMENT_IS_WASM_WORKER
       'mem': wasmMemory,
 #else
       'wasm': wasmModule,
-      'js': Module['mainScriptUrlOrBlob'] || _scriptName,
+      'js': URL.createObjectURL(Module["mainScriptUrlOrBlob"]) || _scriptName,
```

Comment:
So this is fix for the case when `mainScriptUrlOrBlob` is a blob?   Is this because blobs cannot be serialized but object URLs can?

Can we add a test for this?  And perhaps make it a separate PR and it seems somewhat orthogonal.